inc/saiprefixcompression.h

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright (c) 2024 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiprefixcompression.h
+ *
+ * @brief   This module defines SAI prefix compression interface
+ */
+
+#if !defined (__SAIPREFIXCOMPRESSION_H_)
+#define __SAIPREFIXCOMPRESSION_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIPREFIXCOMPRESSION SAI - Prefix Compression API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute Id for SAI prefix compression object
+ */
+typedef enum _sai_prefix_compression_table_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_PREFIX_COMPRESSION_TABLE_ATTR_START,
+
+    /**
+     * @brief Label attribute used to unique identify Table.
+     *
+     * @type char
+     * @flags CREATE_AND_SET
+     * @default ""
+     */
+    SAI_PREFIX_COMPRESSION_TABLE_ATTR_LABEL = SAI_PREFIX_COMPRESSION_TABLE_ATTR_START,
+
+    /**
+     * @brief Prefix Compression table default entry BINCODE.
+     * BINCODE applied to any address that does not match a prefix in the table
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_PREFIX_COMPRESSION_TABLE_ATTR_DEFAULT_ENTRY_BINCODE,
```

Comment:
Can't this be modelled by creating a prefix entry with a prefix length of 0?

Reply:
This attribute allows the user to specific what Bincode they want to associate with the default entry which is a pefix/mask of zero. The default BINCODE value does not have to be zero just the prefix/mask value.

Reply:
Per discussion today will make changes as requested. I will remove this attribute

Reply:
@rck-innovium I removed the default entry attribute, please close if you are ok with the changes